examples/fabric-admin/rpc/RpcServer.cpp

```diff
@@ -62,6 +63,43 @@ class FabricAdmin final : public rpc::FabricAdmin
         return pw::OkStatus();
     }
 
+    pw::Status ReverseCommissionBridge(const chip_rpc_DeviceCommissioningInfo & request, pw_protobuf_Empty & response) override
+    {
+        char saltHex[Crypto::kSpake2p_Max_PBKDF_Salt_Length * 2 + 1];
+        Encoding::BytesToHex(request.salt.bytes, request.salt.size, saltHex, sizeof(saltHex), Encoding::HexFlags::kNullTerminate);
+
+        ChipLogProgress(NotSpecified, "Received ReverseCommissionBridge request");
+
+        SetupPayload setupPayload = SetupPayload();
+
+        setupPayload.setUpPINCode = request.setup_pin;
+        setupPayload.version      = 0;
+        setupPayload.rendezvousInformation.SetValue(RendezvousInformationFlag::kOnNetwork);
+
+        SetupDiscriminator discriminator{};
+        discriminator.SetLongValue(request.discriminator);
+        setupPayload.discriminator = discriminator;
+
+        char payloadBuffer[kMaxManaulCodeLength + 1];
+        MutableCharSpan manualCode(payloadBuffer);
+
+        CHIP_ERROR error = ManualSetupPayloadGenerator(setupPayload).payloadDecimalStringRepresentation(manualCode);
+        if (error == CHIP_NO_ERROR)
+        {
+            NodeId nodeId = DeviceMgr().GetNextAvailableNodeId();
+
+            usleep(kCommissionPrepareTimeMs * 1000);
+
+            DeviceMgr().PairRemoteDevice(nodeId, payloadBuffer);
```

Comment:
Inside `PairRemoteDevice` you are calling `chip-tool pairing code `but you are not filtering for vendor and product id. I don't have a solution for you, but this will prevent you from passing CCTRL_2_2 because DUT is expected to fail as per my understanding of the spec

Reply:
Cecille let me know how we should be doing this, we need to first establish a PASE session and read the BasicInfo cluster and make sure all the values match. If the match we should then do what we are doing here

Reply:
This requires spec text fix first

https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/10259